wodles/aws/aws_s3.py

```diff
@@ -3415,6 +3425,190 @@ def purge_db(self, log_group):
+
+    def __init__(self, access_key: str = None, secret_key: str = None, aws_profile: str = None, **kwargs):
+        WazuhIntegration.__init__(self, access_key=access_key, secret_key=secret_key,
+                                  aws_profile=aws_profile, service_name='s3', **kwargs)
+
+    def obtain_information_from_parquet(self, bucket_path: str, parquet_path: str) -> list:
+        """
+        Fetch a parquet file from a bucket and obtain a list of the events it contains
+
+        Parameters
+        ----------
+        bucket_path : str
+            Path of the bucket to get the parquet file from
+
+        parquet_path : str
+            Relative path of the parquet file inside the bucket
+
+        Returns
+        -------
+        A list with the events contained inside the parquet file.
+        """
+
+        debug(f'Processing file {parquet_path} in {bucket_path}', 2)
+        events = []
+        try:
+            raw_parquet = io.BytesIO(self.client.get_object(Bucket=bucket_path, Key=parquet_path)['Body'].read())
+        except Exception as e:
+            debug(f'Could not get the parquet file {parquet_path} in {bucket_path}: {e}', 1)
+            sys.exit(20)
+        pfile = pq.ParquetFile(raw_parquet)
+        for i in pfile.iter_batches():
+            for j in i.to_pylist():
+                events.append(json.dumps(j))
+        debug(f'Found {len(events)} events in file {parquet_path}', 2)
+        return events
+
+    def process_file(self, message: dict) -> None:
+        """
+        Parse an SQS message, obtain the events associated, and send them to AnalysisD.
+
+        Parameters
+        ----------
+        message : dict
+            An SQS message recieved from the queue
+        """
+        events_in_file = self.obtain_information_from_parquet(bucket_path=message['bucket_path'],
+                                                              parquet_path=message['parquet_path'])
+        for event in events_in_file:
+            self.send_msg(event, dump_json=False)
+        debug(f'{len(events_in_file)} events sent to AnalysisD', 2)
+
+
+class AWSSQSQueue(WazuhIntegration):
+    """
+    Class for getting AWS SQS Queue notifications.
+
+    Attributes
+    ----------
+    name: str
+        Name of the SQS Queue.
+    access_key : str
+        AWS access key id.
+    secret_key : str
+        AWS secret access key.
+    aws_profile : str
+        AWS profile.
+    iam_role_arn : str
+        IAM Role.
+    """
+
+    def __init__(self, name: str, access_key: str = None, secret_key: str = None,
+                 aws_profile: str = None, **kwargs):
+        self.sqs_name = name
+        WazuhIntegration.__init__(self, access_key=access_key, secret_key=secret_key,
+                                  aws_profile=aws_profile, service_name='sqs', **kwargs)
+        self.sts_client = self.get_sts_client(access_key, secret_key, aws_profile)
+        self.account_id = self.sts_client.get_caller_identity().get('Account')
```

Comment:
just to make sure, would this work if IAM roles are used instead?

Reply:
This does work